interactions.go

```diff
@@ -7,8 +7,230 @@ import (
+}
+
+// ApplicationCommandOptionType is type of an slash-command's option.
+type ApplicationCommandOptionType uint8
+
+// Application command option types.
+const (
+	ApplicationCommandOptionSubCommand = ApplicationCommandOptionType(iota + 1)
+	ApplicationCommandOptionSubCommandGroup
+	ApplicationCommandOptionString
+	ApplicationCommandOptionInteger
+	ApplicationCommandOptionBoolean
+	ApplicationCommandOptionUser
+	ApplicationCommandOptionChannel
+	ApplicationCommandOptionRole
+)
+
+// ApplicationCommandOption is representing an option/subcommand/subcommands group.
+type ApplicationCommandOption struct {
+	Type        ApplicationCommandOptionType `json:"type"`
+	Name        string                       `json:"name"`
+	Description string                       `json:"description,omitempty"`
+	// Default     bool                              `json:"default"`
+	Required bool                              `json:"required"`
+	Choices  []*ApplicationCommandOptionChoice `json:"choices"`
+	Options  []*ApplicationCommandOption       `json:"options"`
+}
+
+// ApplicationCommandOptionChoice is representing slash-command's option choice.
+type ApplicationCommandOptionChoice struct {
+	Name  string      `json:"name"`
+	Value interface{} `json:"value"`
+}
+
+// InteractionType is representing interaction type.
+type InteractionType uint8
+
+// Interaction types
+const (
+	InteractionPing = InteractionType(iota + 1)
+	InteractionApplicationCommand
+)
+
+// Interaction is representing interaction with application.
+type Interaction struct {
+	ID        string                            `json:"id"`
+	Type      InteractionType                   `json:"type"`
+	Data      ApplicationCommandInteractionData `json:"data"`
+	GuildID   string                            `json:"guild_id"`
+	ChannelID string                            `json:"channel_id"`
+	Member    *Member                           `json:"member"`
+	Token     string                            `json:"token"`
+	Version   int                               `json:"version"`
+}
+
+// ApplicationCommandInteractionData is representing interaction data for application command.
+type ApplicationCommandInteractionData struct {
+	ID      string                                     `json:"id"`
+	Name    string                                     `json:"name"`
+	Options []*ApplicationCommandInteractionDataOption `json:"options"`
+}
+
+// ApplicationCommandInteractionDataOption is representing an option of application's command.
+type ApplicationCommandInteractionDataOption struct {
+	Name string `json:"name"`
+	// Contains the value specified by InteractionType
+	Value   interface{}                                `json:"value,omitempty"`
+	Options []*ApplicationCommandInteractionDataOption `json:"options,omitempty"`
+}
+
+// IntValue is utility function for casting option value to integer
+func (o ApplicationCommandInteractionDataOption) IntValue() (i int64) {
+	if v, ok := o.Value.(float64); ok {
```

Comment:
Shouldn't the condition be `o.Value.(int64)` and not `o.Value.(float64)`? You want to make sure that the underlying type of the interface is an `int64`.

Reply:
One question: did you read the `encoding/json` package documentation?
I personally don't see any `int64` here.
> To unmarshal JSON into an interface value, Unmarshal stores one of these in the interface value:
> ```
> bool, for JSON booleans
> float64, for JSON numbers
> string, for JSON strings
> []interface{}, for JSON arrays
> map[string]interface{}, for JSON objects
> nil for JSON null
> ```

Reply:
Please be respectful fedor. No need to be like this.

Reply:
I'm just upset, because he didn't care about reading the `encoding/json` package's documentation

Reply:
That does not give you any right to be disrespectful and snarky like that. It was an honest mistake that anyone could've made. It's not an immediately obvious thing, it requires knowledge of the fact that javascript only has floats and by extension the same applies to JSON. it's completely understandable that someone might think to use int64 here if they don't know this. You must understand this and be patient and respectful 🙂

Reply:
Well, okay, but I didn't tried to be disrespectful, just pointed to the docs and asked him about reading them, nothing more. Also, I don't see any point to continue this conversation, situation is resolved.